### PR TITLE
Convert I2C controller drivers to use new DT device macros

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

**Comment by galak** [comment]
This is based on PR #30400

**Comment by galak** [comment]
Marked blocked on PR #30400

**Review by scottwcpg — APPROVED** [review]

**Review by sjg20 — APPROVED** [review]

File: drivers/i2c/i2c_sbcon.c [hunk 1]
```diff
@@ -116,8 +116,9 @@ static const struct i2c_sbcon_config i2c_sbcon_dev_cfg_##_num = {	\
 	.sbcon		= (void *)DT_INST_REG_ADDR(_num), \
 };									\
 									\
-DEVICE_AND_API_INIT(i2c_sbcon_##_num, DT_INST_LABEL(_num), \
+DEVICE_DT_INST_DEFINE(num,						\
 	    i2c_sbcon_init,						\
+	    NULL,							\
```
Comment (pabigot) [inline drivers/i2c/i2c_sbcon.c:121]: The documented parameter when power management is not supported is `device_pm_control_nop`.  Let's please use that until we decide is unnecessary and make everything consistent with intent.

**Review by pabigot — CHANGES_REQUESTED** [review]

File: drivers/i2c/i2c_sbcon.c (hunk 1 above)
Comment (galak) [reply drivers/i2c/i2c_sbcon.c:121]: Ok, will update the other PRs to match.

**Review by MaureenHelm — APPROVED** [review]

**Review by pabigot — APPROVED** [review]

**Review by nandojve — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**